### PR TITLE
Additional procedures for bloom fulltext

### PR DESCRIPTION
**Pull request by ragadeeshu**

### CODE REVIEW

File: enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/integrations/bloom/BloomProcedures.java [hunk 1]
```diff
@@ -55,41 +59,89 @@ public void awaitPopulation() throws IOException
         provider.awaitPopulation();
     }
 
+    @Description( "Returns the property keys indexed by the bloom addon" )
+    @Procedure( name = "db.fulltext.bloomFulltextProperties", mode = READ )
+    public Stream<PropertyOutput> bloomFulltextProperties() throws Exception
+    {
+        return provider.getProperties(BLOOM_NODES, NODES).stream().map( PropertyOutput::new );
```
Comment (chrisvest) [inline enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/integrations/bloom/BloomProcedures.java]: Checkstyle spaces.

Comment (ragadeeshu) [reply enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/integrations/bloom/BloomProcedures.java]: Fixed

**Comment by ragadeeshu** [comment]
@chrisvest updated

File: enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java [hunk 2]
```diff
@@ -220,6 +222,28 @@ public ReadOnlyFulltext getReader( String identifier, FulltextIndexType type ) t
         }
     }
 
+    public Set<String> getProperties( String identifier, FulltextIndexType type )
+    {
+        return applyToMatchingIndex( identifier, type, LuceneFulltext::getProperties );
+    }
+
+    private <E> E applyToMatchingIndex( String identifier, FulltextIndexType type, Function<LuceneFulltext,E> getProperties )
```
Comment (chrisvest) [inline enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java]: This `getProperties` function should have a more generic name, I think.

File: enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/LuceneFulltext.java [hunk 3]
```diff
@@ -34,13 +34,15 @@
...
 import org.neo4j.kernel.api.impl.schema.writer.PartitionedIndexWriter;
+import org.neo4j.kernel.api.index.InternalIndexState;
 
 class LuceneFulltext extends AbstractLuceneIndex
 {
     private final Analyzer analyzer;
     private final String identifier;
     private final FulltextProvider.FulltextIndexType type;
     private final Set<String> properties;
+    private InternalIndexState state;
```
Comment (chrisvest) [inline enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/LuceneFulltext.java]: Perhaps this needs to be volatile, just to be safe?

**Review by chrisvest — COMMENTED** [review]
I had a couple of comments. If you just fix those up, it's good to go in, I think. 👍

File: enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java (hunk 2 above)
Comment (ragadeeshu) [reply enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java]: whoops

**Review by chrisvest — APPROVED** [review]

**Comment by jexp** [comment]
Why do they have fulltext twice in the name? and bloom?
Perhaps align with the Cypher team on naming like they did for the explicit indexes, so it is consistent.
/cc @craigtaverner 

db.fulltext.bloomFulltextStatus -> db.fulltext.status

**Comment by chrisvest** [comment]
@jexp It's not meant to be a user-facing feature. It's only for the bloom project, so we don't to contaminate the namespace if we want to do a fully featured, general purpose fulltext index later.

**Pull request merged**